### DISCUSSION TITLE
System.InvalidOperationException: 'Visual was invalidated during the render pass'

### DISCUSSION THREAD

**Discussion by sonicsmooth** [AvaloniaUI/Avalonia#9617] [Q&A]
Using Windows 10 Home, Avalonia 11.0.0.0-preview4, Visual Studio Community 17.4.2

I'm following the AngelSix youtube video 009 https://www.youtube.com/watch?v=T1kErI5VkW4
At 12:40 he introduces the following line in MainView.axaml.cs:
`mChannelConfigPopup.Margin = new Thickness(0,0,0,0);`

This builds fine but I get the exception upon assigning a new value to Margin (creating the Thickness object works okay):
![image](https://user-images.githubusercontent.com/672429/205519564-ad750a6a-267a-4094-bbf2-bbea43d8c87b.png)

I traced this back to some `if(_updating)` statement in CompositingRenderer.cs/`AddDirty(IVisual visual)`, which doesn't help me much as I'm pretty new to this.

I couldn't find this issue on google or stackoverflow.
thanks.

**Comment by maxkatz6** [comment]
Answered in the issue https://github.com/AvaloniaUI/Avalonia/issues/9618#issuecomment-1336653052
